bin/python_modules/integronfinder_process.py

```diff
@@ -9,42 +9,64 @@
-                                        mge_counter += 1
-                                        mge_id = "inf_" + str(mge_counter)
-                                        start = int(feature.location.start)
-                                        end = int(feature.location.end)
-                                        coord = (start, end)
-                                        value = (id_replicon, description, coord)
-                                        mge_data[mge_id] = value
-                                if feature.type == "attC":
-                                    if flag == 1:
-                                        start = str(feature.location.start)
-                                        end = str(feature.location.end)
-                                        coord = (start, end)
-                                        attC_site[mge_id] = coord
-                                        flag = 0
+                    gbk_files_list.append(gbk_file)
+
+
+    mge_counter = 0
+    attC_site = {}
+    for gbk_file in gbk_files_list:
+        for gb_record in SeqIO.parse(gbk_file, "genbank"):
+
+            # Get indexes for integron records and attC
+            indexes_integron, indexes_attc = [], []
+            for index, feature in enumerate(gb_record.features):
+                if feature.type == "integron":
+                    indexes_integron.append(index)
+                elif feature.type == "attC":
+                    indexes_attc.append(index)
+
+            # process all integrons
+            for index_integron in indexes_integron:
+                integron_feature = gb_record.features[index_integron]
+                if integron_feature.qualifiers["integron_type"][0] == "complete":
+                    mge_counter += 1
+                    mge_id = "inf_" + str(mge_counter)
+                    mge_start = int(integron_feature.location.start)
+                    if mge_start == 0:
+                        mge_start = 1
+                    mge_end = int(integron_feature.location.end)
+                    mge_coord = (mge_start, mge_end)
+                    description = "mobile_element_type=complete_integron"
+                    id_replicon = gbk_file.replace('.gbk','')
+                    value = (id_replicon, description, mge_coord)
+                    mge_data[mge_id] = value
+
+                    print('process integron -> check attC')
+                    print(value)
+                    # check attC
+                    for index_attc in indexes_attc:
+                        attc_feature = gb_record.features[index_attc]
+                        attc_start = str(attc_feature.location.start)
+                        if attc_start == 0:
+                            attc_start = 1
+                        attc_end = str(attc_feature.location.end)
+                        attc_coord = (attc_start, attc_end)
+
+                        if mge_id in attC_site:
```

Comment:
Martin's favourite: replace whole if-else to setdefault magic 
```
attC_site.setdefault(mge_id, [])
attC_site[mge_id].append(attc_coord)
```

Reply:
Done!